apps/epic-react/src/lib/products.ts

```diff
@@ -52,6 +52,78 @@ export async function getProduct(productId: string): Promise<Product | null> {
   return ProductSchema.parse(product)
 }
 
+export const getProductBySlug = async (productSlug: string) => {
+  const product = await sanityClient.fetch(
+    groq`*[_type == 'product' && slug.current == $productSlug][0] {
+  "name": title,
+  "slug": slug.current,
+  productId,
+  description,
+  state,
+  type,
+  title,
+  body,
+  _createdAt,
+  action,
+  "welcomeVideo": welcomeVideo->{"muxPlaybackId":muxAsset.muxPlaybackId, poster},
+  image {
+    url,
+    alt
+  },
+  ogImage,
+  modules[]->{
+    "slug": slug.current,
+    "instructors": contributors[@.role == 'instructor'].contributor->{
+              ...,
+              "slug": slug.current,
+          },
+    moduleType,
+    title,
+    "image": {
+      "url": image.url
+    },
+    state,
+    "lessons": resources[@->._type in ['exercise', 'explainer', 'lesson', 'interview']]->{
+      ..., "slug": slug.current},
```

Comment:
Also I've noticed that in remote version of `getProductBySlug` there is the following code
`{"slug": slug.current,...}` instead of `{..., "slug": slug.current}`. That means that in fetched object we still get `slug: {current: XXX}` instead of `{slug: XXX}`